flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -169,8 +181,18 @@ protected Table createHiveTable(ObjectPath tablePath, CatalogBaseTable table) {
 				sd.setCols(allColumns);
 				hiveTable.setPartitionKeys(new ArrayList<>());
 			}
+		} else if (table instanceof CatalogView) {
```

Comment:
should we check against HiveCatalogView instead, as we are casting table to it right after this line? Same applies to the check in the if condition.